File: src/core/messaging/MessageBus.cpp
// src/core/messaging/MessageBus.cpp

#include "core/messaging/MessageBus.hpp"

namespace XAlgo {

MessageBus::MessageBus() = default;

void MessageBus::publish(std::shared_ptr<MarketEvent> event) {
    if (callback_) {
        callback_(std::move(event));
    }
}

} // namespace XAlgo

File: src/messaging/message_bus.py
# /src/messaging/message_bus.py

from typing import Callable, List

class MessageBus:
    def __init__(self):
        self.subscribers: List[Callable] = []

    def subscribe(self, callback: Callable):
        self.subscribers.append(callback)

    def publish(self, event):
        for callback in self.subscribers:
            callback(event)
